src/panels/config/zha/zha-node.ts

```diff
@@ -0,0 +1,309 @@
+import "@polymer/iron-flex-layout/iron-flex-layout-classes";
+import "../../../components/buttons/ha-call-service-button";
+import "../../../components/ha-service-description";
+import "../../../resources/ha-style";
+import "../ha-config-section";
+
+import { HomeAssistant } from "../../../types";
+import { HassEntity } from "home-assistant-js-websocket";
+import computeStateName from "../../../common/entity/compute_state_name";
+import sortByName from "../../../common/entity/states_sort_by_name";
+import { fireEvent, HASSDomEvent } from "../../../common/dom/fire_event";
+import {
+  ItemSelectedEvent,
+  NodeServiceData,
+  ZHAEntitySelectedParams,
+} from "./types";
+
+import "./zha-entities";
+import "./zha-clusters";
+
+declare global {
+  // for fire event
+  interface HASSDomEvents {
+    "zha-node-selected": {
+      node?: HassEntity;
+    };
+  }
+}
+
+export class ZHANode extends LitElement {
+  public hass?: HomeAssistant;
+  public isWide?: boolean;
+  private _showHelp: boolean;
+  private _selectedNodeIndex: number;
+  private _selectedNode?: HassEntity;
+  private _selectedEntity?: HassEntity;
+  private _serviceData?: {};
+  private _haStyle?: DocumentFragment;
+  private _ironFlex?: DocumentFragment;
+  private _nodes: HassEntity[];
+
+  constructor() {
+    super();
+    this._showHelp = false;
+    this._selectedNodeIndex = -1;
+    this._nodes = [];
+  }
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      isWide: {},
+      _showHelp: {},
+      _selectedNodeIndex: {},
+      _selectedNode: {},
+      _serviceData: {},
+      _selectedEntity: {},
+    };
+  }
+
+  protected render(): TemplateResult {
+    this._nodes = this._computeNodes(this.hass);
+    return html`
+      ${this.renderStyle()}
+      <ha-config-section .isWide="${this.isWide}">
+        <div style="position: relative" slot="header">
+          <span>Node Management</span>
+          <paper-icon-button
+            class="toggle-help-icon"
+            @click="${this._onHelpTap}"
+            icon="hass:help-circle"
+          ></paper-icon-button>
+        </div>
+        <span slot="introduction">
+          Run ZHA commands that affect a single node. Pick a node to see a list
+          of available commands. <br /><br />Note: Sleepy (battery powered)
+          devices need to be awake when executing commands against them. You can
+          generally wake a sleepy device by triggering it. <br /><br />Some
+          devices such as Xiaomi sensors have a wake up button that you can
+          press at ~5 second intervals that keep devices awake while you
+          interact with them.
+        </span>
+        <paper-card class="content">
+          ${this._renderNodePicker()}
+          <div ?hidden="${!this._showHelp}" style="color: grey; padding: 16px">
+            Select node to view per-node options
+          </div>
+          ${this._selectedNodeIndex !== -1 ? this._renderNodeActions() : ""}
```

Comment:
I do kind of like extracting the rendering that is within a ternary...anyone else want to weigh in on this? Does make it easier to see what is happening IMO

Reply:
This is gonna be slippery :) am I leaving these or removing them?

Reply:
I left these as is because this template is a bit larger than the rest and I agree it makes the ternary statements easier to read / reason about